### DISCUSSION TITLE
Support for alternative online meeting providers like Telegram?

### DISCUSSION THREAD

**Discussion by Strompf** [code4recovery/12-step-meeting-list#1230] [Q&A] [answered]
How to incorporate online meetings that don't use any of the default providers? E.g.:

- Meetings on Telegram?
- Meetings that publish their link through a URL shortener like Bit.ly?

It seems that whenever I try to add such a URL to the field "URL", that it isn't accepted.
If this isn't possible, maybe a suggestion for a workaround?

Thanks already & regards,
Jeroen Strompf

**Comment by kiyote33** [comment, marked as answer]
Hi Jeroen (@Strompf), Yes we do have a work-around for adding an online meeting provider which is not already in our online providers list. It works by adding some code to a child theme which might look something like this:
`if (!empty($tsml_conference_providers)) {
	$tsml_conference_providers += [
		'[your_telegram_URL' => 'Telegram',
	];
}`.
Long term, it might be best if we add Telegram to our internal provider listing in a future release. What does an example Telegram URL look like (eg https://telegram.online.meeting.us/join/12356789?pwd=12345)?

**Reply by Strompf** [reply]
Thank you, @kiyote33 - It works! I'm so relieved to see that we can add conference providers ourselves in an easy way. What I added to functions.php:

```
////////////////////////////////////////////////////////////////////////////////
// Add Telegram as Conference Provider
////////////////////////////////////////////////////////////////////////////////
//
if (!empty($tsml_conference_providers))
{
    $tsml_conference_providers += [ 
        't.me' => 'Telegram',
    ];
}
```
I was quite puzzled that I couldn't find this info anywhere, so just in case I'm not the only one with this issue, here is in more details how I did this:

I first searched for the definition of the array ```$tsml_conference_providers``` through Bash with ```grep -rn . -e "tsml_conference_providers"```. I found the definition in file ```wp-content/plugins/12-step-meeting-list/includes/variables.php``` around lines 38-54 (Oct. 2023):
```
//list of valid conference providers (matches Meeting Guide app). set this to null in your theme if you don't want to validate
$tsml_conference_providers = [
	'bluejeans.com' => 'Bluejeans',
	'discord.gg' => 'Discord',
	'freeconference.com' => 'Free Conference',
	'freeconferencecall.com' => 'FreeConferenceCall',
	'goto.com' => 'GoTo',
	'gotomeet.me' => 'GoTo',
	'gotomeeting.com' => 'GoTo',
	'meet.google.com' => 'Google Hangouts',
	'meet.jit.si' => 'Jitsi',
	'meetings.dialpad.com' => 'Dialpad',
	'skype.com' => 'Skype',
	'webex.com' => 'WebEx',
	'zoho.com' => 'Zoho',
	'zoom.us' => 'Zoom',
];
```
As you can see, the pattern seems to be ```'root URL' => 'Name'```, which is ```'t.me' ⇒ 'Telegram'``` in our case - I only have to add two meetings with the same Telegram URL. I have no idea if there are also other Telegram URLs, just as there are three kinds of GoTO URLs in the code above.

I added the lines mentioned at the beginning of this comment to the ```function.php``` file of the child theme on our site, which in our case happens to be: ```wp-content/themes/enfold-child/functions.php```

Lastly: This addition to ```functions.php``` only became active after first leaving the current meeting object in WordPress - Just refreshing the screen somehow wasn't enough.

Details: https://wiki.devliegendebrigade.nl/TSML_(WordPress_12_Step_Meeting_List_plugin)#Add_a_conference_provider

**Reply by kiyote33** [reply]
Congrats! So glad to hear it worked out for you.

**Comment by Strompf** [comment]
BTW: Concerning meeting URLs published through 'URL shorteners' like Bit.ly: Just use the original 'expanded' URL in the URL field, and use the shortened URL in the URL Notes, or wherever you want to include it
